src/movies.js

```diff
@@ -1,16 +1,63 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(movies) {
+const result = movies.map(function(names) {
+    return names.director;
+})
+    return result;
+}
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies) {
+    const dramaMovies = movies.filter(function(number){
+        if(number.director === 'Steven Spielberg' && number.genre.indexOf('Drama') !== -1){
+            return true;
+        } else {
+            return false;
+        }
+    })
+    return dramaMovies.length
+}
+
+//need to use indeof = check if the element you are looking for is in the array
+//
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
-// Iteration 4: Drama movies - Get the average of Drama Movies
+function ratesAverage(movies){
+    if(movies.length === 0) return 0;
+      const getTheSum = movies.reduce(function(sum,movie){
+        if(!movie.rate) return sum;
+        return sum + movie.rate
+      }, 0)
+      const getTheAve = getTheSum / movies.length
```

Comment:
I would simply call this variable "average"/"avg". Names with a verb usually refer to a method/function.